src/components/composite/DepositFlow/DepositFlow.tsx

```diff
@@ -60,6 +105,15 @@ export const DepositFlowOnDemandButton: FC<
     if (!earnOpportunity) {
       return;
     }
+    trackEvent({
+      category: TrackingCategory.Earn,
+      action: TrackingAction.ClickEarnDepositButton,
+      label: 'click-earn-deposit-button',
+      data: {
+        [TrackingEventParameter.EarnOpportunitySlug]:
+          earnOpportunity.slug || '',
+      },
+    });
```

Comment:
Same as above, especially here, I suppose those line are supposed to changed together with the one above ([code that changes together should stay together](https://codecraft.medium.com/segregation-of-responsibilities-with-the-single-responsibility-principle-427bf2faa0a9#:~:text=Classes%20that%20change%20together%20belong%20together.%20Classes%20that%20are%20not%20reused%20together%20should%20not%20be%20grouped.))

```suggestion
    trackEarnDeposit(earnOpportunity.slug)
```